main.ts

```diff
@@ -104,7 +106,41 @@ function fixBibleReferences(app) {
   }
 
   let wikiBible = function(b: { book: any; chapter: any; verse: any; ref: any; }) {
-    return `[[ESV/${b.book}/${b.book}-${b.chapter}${prefix(b.verse, "#")}|${b.ref}]]`;
+    /*
+      Replaces a custom pattern like "{{book}}/{{book}}-{{chapter}}#{{verse}}|{{input}}"
+      with the variables stored.
+    */
+     
+    // There is surely a prettier way to do this – maybe objects?
+    let toReplace = ["{{book}}", "{{chapter}}", "{{verse}}", "{{input}}"]
+    let replaceWith = [`${b.book}`, `${b.chapter}`, `${b.verse}`, `${b.ref}`]
+    
+
+    for (let i = 0; i <= toReplace.length; i++){
+      if (i === 0) {
+        // For the first iteration, pull the custom pattern in the settings
+        var linkPattern = this.plugin.settings.customLinkScheme;
+      }
+      // Replace custom patterns with the variables
+      var linkPattern:any = linkPattern.replaceAll(toReplace[i], replaceWith[i])
+    }

```

Comment:
Nice! There may be a cleaner way to do this, but as my professor used to say:

1. Make it work.
2. Make it nice.
3. Make it fast.